TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Eel/FlowQueryOperations/ParentsUntilOperation.php

```diff
@@ -64,16 +64,14 @@ public function evaluate(FlowQuery $flowQuery, array $arguments)
             $siteNode = $contextNode->getContext()->getCurrentSiteNode();
             $parentNodes = $this->getParents($contextNode, $siteNode);
 
-            if (isset($arguments[0]) && !empty($arguments[0])) {
-                $untilQuery = new FlowQuery($parentNodes);
-                $untilQuery->pushOperation('filter', array($arguments[0]));
-
+            if (isset($arguments[0]) && !empty($arguments[0] && isset($parentNodes[0]))) {
+                $untilQuery = new FlowQuery(array($parentNodes[0]));
+                $untilQuery->pushOperation('closest', array($arguments[0]));
                 $until = $untilQuery->get();
             }
 
-            if (isset($until) && !empty($until)) {
-                $until = end($until);
-                $parentNodes = $this->getNodesUntil($parentNodes, $until);
+            if (isset($until[0]) && !empty($until)) {
```

Comment:
This change seems wrong. !empty() should be checked first. Additionally it may make sense to check if it is an array and not empty.